### PR TITLE
Prevent DSS client session expiration in `copy_bundles.py` (#1325)

### PR DESCRIPTION
**Pull request by danielsotirhos**

### CODE REVIEW

File: scripts/copy_bundles.py [hunk 1]
```diff
@@ -90,24 +92,31 @@ def _parse_args(cls, argv):
         args = parser.parse_args(argv)
         return args
 
-    num_workers = 32
+    num_workers = 8
```
Comment (hannes-ucsc) [inline scripts/copy_bundles.py]: That we are not sure about just yet. Still collecting data.

File: scripts/copy_bundles.py [hunk 2]
```diff
@@ -90,24 +92,31 @@ def _parse_args(cls, argv):
...
-                                 adapter_args=dict(pool_maxsize=self.num_workers))
+        self._destination = None
+        self._destination_expiration = None
+        self._destination_lock = RLock()
+
+    @property
+    def destination(self):
+        with self._destination_lock:
+            if self._destination is None or self._destination_expiration < time.time():
+                self._destination = config.dss_client(dss_endpoint=urlunparse(self.args.destination),
```
Comment (hannes-ucsc) [inline scripts/copy_bundles.py]: Let's add a INFO log message here, sth like "Allocating new DSS client for <dss_endpoint> to expire in … seconds."

**Comment by danielsotirhos** [comment]
Results from using copy_bundles script to copy projects related to portal integrations from `prod` to `integration`:

https://data.humancellatlas.org/explore/projects/c4077b3c-5c98-4d26-a614-246d12c2e5d7
https://integration.data.humancellatlas.org/explore/projects/c4077b3c-5c98-4d26-a614-246d12c2e5d7
Copy with no prefix
Bundle Count: 14
Total Files (prod): 82
Total Files (int) : 82

https://data.humancellatlas.org/explore/projects/90bd6933-40c0-48d4-8d76-778c103bf545
https://integration.data.humancellatlas.org/explore/projects/90bd6933-40c0-48d4-8d76-778c103bf545
Copy with no prefix
Bundle Count: 2,244
Total Files (prod): 2,244
Total Files (int) : 2,244

https://data.humancellatlas.org/explore/projects/cddab57b-6868-4be4-806f-395ed9dd635a
https://integration.data.humancellatlas.org/explore/projects/cddab57b-6868-4be4-806f-395ed9dd635a
Copy with no prefix (canceled), then Copy with prefix 16 times [0-f]
Bundle Count: 5,087
Total Files (prod): 50,880
Total Files (int) : 50,862

https://data.humancellatlas.org/explore/projects/f83165c5-e2ea-4d15-a5cf-33f3550bffde
https://integration.data.humancellatlas.org/explore/projects/f83165c5-e2ea-4d15-a5cf-33f3550bffde
Copy with prefix 0
Bundle Count: 15,252
Total Files (prod): 152,281
Total Files (int) : 10,453

https://data.humancellatlas.org/explore/projects/2043c65a-1cf8-4828-a656-9e247d4e64f1
https://integration.data.humancellatlas.org/explore/projects/2043c65a-1cf8-4828-a656-9e247d4e64f1
Copy with no prefix
Bundle Count: 3,465
Total Files (prod): 34,660
Total Files (int) : 22,798

https://data.humancellatlas.org/explore/projects/ae71be1d-ddd8-4feb-9bed-24c3ddb6e1ad
https://integration.data.humancellatlas.org/explore/projects/ae71be1d-ddd8-4feb-9bed-24c3ddb6e1ad
Copy with no prefix
Bundle Count: 3,514
Total Files (prod): 3,514
Total Files (int) : 3,477

https://data.humancellatlas.org/explore/projects/e0009214-c0a0-4a7b-96e2-d6a83e966ce0
https://integration.data.humancellatlas.org/explore/projects/e0009214-c0a0-4a7b-96e2-d6a83e966ce0
Copy with prefix 0
Bundle Count: 99,840
Total Files (prod): 199,680
Total Files (int) : 10,942

**Comment by hannes-ucsc** [comment]
So apparently we still get dropped notifications even with just 8 workers?

**Pull request merged**